arcgis-ios-sdk-samples/Maps/Generate offline map (overrides)/OfflineMapParameterOverridesViewController.swift

```diff
@@ -0,0 +1,263 @@
+    }
+    
+    /// Updates the `AGSGenerateOfflineMapParameterOverrides` object with the user-set values.
+    private func setParameterOverridesFromUI(){
+        
+        restrictBasemapScaleLevelRange()
+        bufferBasemapAreaOfInterest()
+        evaluateLayerVisiblity()
+        addHydrantFilter()
+        evaluatePipeLayersExtentCropping()
+    }
+    
+    //MARK: - Basemap adjustment
+    
+    private func restrictBasemapScaleLevelRange() {
+        
+        /// The user-set min scale value
+        let minScale = Int(minScaleLevelSlider.value)
+        /// The user-set max scale value
+        let maxScale = Int(maxScaleLevelSlider.value)
+        
+        guard let tileCacheParameters = getExportTileCacheParametersForBasemapLayer(),
+            // Ensure that the lower bound of the range is not greater than the upper bound
+            minScale <= maxScale else {
+                return
+        }
+        
+        let scaleLevelRange = minScale...maxScale
+        let scaleLevelIDs = Array(scaleLevelRange) as [NSNumber]
+        // Override the default level IDs
+        tileCacheParameters.levelIDs = scaleLevelIDs
+    }
+    
+    private func bufferBasemapAreaOfInterest() {
+        
+        guard let tileCacheParameters = getExportTileCacheParametersForBasemapLayer(),
+            /// The area initially specified for download when the default parameters object was created
+            let areaOfInterest = tileCacheParameters.areaOfInterest else{
+                return
+        }
+        
+        /// The user-set distance value
+        let basemapExtentBufferDistance = Double(basemapExtentBufferSlider.value)
+        
+        // Assuming the distance is positive, expand the downloaded area by the given amount
+        let bufferedArea = AGSGeometryEngine.bufferGeometry(areaOfInterest, byDistance: basemapExtentBufferDistance)
+        // Override the default area of interest
+        tileCacheParameters.areaOfInterest = bufferedArea
+    }
+    
+    //MARK: - Layer adjustment
+    
+    private func addHydrantFilter(){
+        
+        /// The user-set min flow rate value
+        let minFlowRate = minHydrantFlowRateSlider.value
+        
+        for option in getGenerateGeodatabaseParametersLayerOptions(forLayerNamed: "Hydrant") {
+            // Set the SQL where clause for this layer's options, filtering features based on the FLOW field values
+            option.whereClause = "FLOW >= \(minFlowRate)"
+        }
+    }
+    
+    private func evaluateLayerVisiblity(){
+        
+        func excludeLayerFromDownload(named name: String) {
+            if let layer = operationalMapLayer(named: name),
+                let serviceLayerID = serviceLayerID(for: layer),
+                let parameters = getGenerateGeodatabaseParameters(forLayer: layer){
+                // Remove the options for this layer from the parameters
+                parameters.layerOptions.removeAll { return $0.layerID == serviceLayerID }
```

Comment:
You know that the `return` keyword isn't needed here, right? It isn't wrong; just unnecessary.

Reply:
Unintentional oversight.